internal/layer2/announcer.go

```diff
@@ -131,24 +137,49 @@ func (a *Announce) updateInterfaces() {
 	return
 }
 
-func (a *Announce) spam(name string) {
-	// TODO: should abort if we lose control of the IP mid-spam.
-	start := time.Now()
-	for time.Since(start) < 5*time.Second {
-		if err := a.gratuitous(name); err != nil {
-			a.logger.Log("op", "gratuitousAnnounce", "error", err, "service", name, "msg", "failed to make gratuitous IP announcement")
+func (a *Announce) spamLoop() {
+	m := map[string]time.Time{}
+	// see https://github.com/metallb/metallb/issues/172 for the 1100 choice
+	ticker := time.NewTicker(1100 * time.Millisecond)
```

Comment:
It would be nice to turn this off when there's no active spamming happening.  Otherwise when it should be otherwise idle, it's still waking up every 1.1s for no reason.

Reply:
might be doable with ticker.Stop() and ticker.Reset(), will have a look

Reply:
ticker.Reset() is golang 1.15, do we want to bump minimum version to latest and greatest ? (even my Fedora only has 1.14)

Reply:
I'll open a issue as soon as we merge this PR so we don't forget